service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GpcFindDocumentsComponentTest.java

```diff
@@ -0,0 +1,114 @@
+import uk.nhs.adaptors.gp2gp.ehr.EhrStatusConstants;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus;
+import uk.nhs.adaptors.gp2gp.testcontainers.ActiveMQExtension;
+import uk.nhs.adaptors.gp2gp.testcontainers.MongoDBExtension;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+@ExtendWith({SpringExtension.class, MongoDBExtension.class, ActiveMQExtension.class})
+@SpringBootTest
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class GpcFindDocumentsComponentTest extends BaseTaskTest {
+    private static final String NHS_NUMBER_WITH_DOCUMENT = "9690937286";
+    private static final String NHS_NUMBER_WITHOUT_DOCUMENT = "9690937294";
+    private static final String VALID_DOCUMENT_URL = "https://orange.testlab.nhs.uk/B82617/STU3/1/gpconnect/documents/fhir/Binary/07a6483f-732b-461e-86b6-edb665c45510";
+
+    @Autowired
+    private GpcFindDocumentsTaskExecutor gpcFindDocumentsTaskExecutor;
+    @Autowired
+    private EhrExtractStatusRepository ehrExtractStatusRepository;
+
+    private EhrExtractStatus setupDatabase() {
+        var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
+        ehrExtractStatus.setGpcAccessDocument(EhrExtractStatus.GpcAccessDocument.builder()
+            .documents(List.of(
+                EhrExtractStatus.GpcAccessDocument.GpcDocument.builder()
+                    .documentId(EhrStatusConstants.DOCUMENT_ID)
+                    .accessDocumentUrl(EhrStatusConstants.GPC_ACCESS_DOCUMENT_URL)
+                    .build()
+            ))
+            .build());
+        return ehrExtractStatusRepository.save(ehrExtractStatus);
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithDocument_Expect_DatabaseToBeUpdated() throws IOException {
+        var ehrExtractStatus = setupDatabase();
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITH_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
+        var updatedEhrExtractStatus = ehrExtractStatusRepository.findByConversationId(taskDefinition.getConversationId()).get();
+        assertThatAccessRecordWasUpdated(updatedEhrExtractStatus, ehrExtractStatus, taskDefinition);
+        Mockito.verify(taskDispatcher).createTask(buildGetDocumentTask(taskDefinition));
+
+    }
+
+    @Test
+    public void When_FindDocumentTaskIsStartedForPatientWithoutDocument_Expect_DatabaseToNotBeUpdated() {
+        var ehrExtractStatus = setupDatabase();
+        var taskDefinition = buildFindDocumentTask(ehrExtractStatus, NHS_NUMBER_WITHOUT_DOCUMENT);
+        gpcFindDocumentsTaskExecutor.execute(taskDefinition);
+
```

Comment:
I think this should assert an error is thrown

Reply:
No error should be thrown at this point.

Reply:
Discussed with Thomas, error not thrown here

Reply:
Make sure we're covering off all the cases:
1) Find a patient: one patient, Find patient's documents: at least one document
2) Find a patient: no patients found (200 empty searchSet)
3) Find patient's documents: no documents found (200 empty searchSet)
4) API Error: any response other than 200. Probably easiest way to to use an invalid NHS number e.g. "ASDF" to get a 400 at "Find a patient"

Reply:
Test added to validate fourth point: API Error: any response other than 200. Probably easiest way to to use an invalid NHS number e.g. "ASDF" to get a 400 at "Find a patient"